platforms/core-configuration/model-core/src/integTest/groovy/org/gradle/api/provider/CollectionPropertyIntegrationTest.groovy

```diff
@@ -473,4 +473,26 @@ task wrongPropertyElementTypeApi {
         failure.assertHasCause("Cannot query the value of task ':thing' property 'prop' because it has no value available.")
     }
 
+    def "circular evaluation of #collection property is detected"() {
+        buildFile """
+            def myCollection = objects.${initializer}(String)
+            def p = myCollection.map { it.join(", ") }
+            myCollection.add(p)
+
+            myCollection.get()
+
+            tasks.register("verify") {}
+        """
+
+        when:
+        fails "verify"
```

Comment:
🧪 I'd call `help` here instead of introducing a new task since the error happens during the configuration phase

Reply:
Turns out that task (if we moved the property evaluation somewhere below the task registration) fails to be registered with:

> Cannot add task 'verify' as a task with that name already exists.